marklogic-sesame/src/main/java/com/marklogic/semantics/sesame/client/TripleWriteCache.java

```diff
@@ -79,8 +79,16 @@ protected synchronized void flush() throws RepositoryException, MalformedQueryEx
                    entireQuery.append(" GRAPH <" + ctx + "> { ");
                }
                 for (Statement stmt : cache.filter(null, null, null, ctx)) {
-                    entireQuery.append("<" + stmt.getSubject().stringValue() + "> ");
-                    entireQuery.append("<" + stmt.getPredicate().stringValue() + "> ");
+                    if (stmt.getSubject() instanceof org.openrdf.model.BNode) {
+                        entireQuery.append("<http://marklogic.com/semantics/blank/" + stmt.getSubject().stringValue() + "> ");
+                    }else {
+                        entireQuery.append("<" + stmt.getSubject().stringValue() + "> ");
```

Comment:
you are probably right, though so far all my tests have been ok - its a pattern I have seen deep in Sesame codebase ... lets discuss alternate approaches soon.